### ISSUE TITLE
emoji domain support

### ISSUE THREAD

**Issue by tycho01** [brave/brave-browser#8208]
## Description 
Non-ascii characters can be represented in domain names using Punycode.
This means there are emoji domains such as [😎🌐.ws](http://😎🌐.ws), but only few browsers still support actually displaying such domains using the intended characters (such as emoji), rather than as Punycode.

## Steps to Reproduce

   1. go to [😎🌐.ws](http://😎🌐.ws)
   2. realize your address bar isn't showing emoji
   3. feel very disappointed

You can also test with [i❤️.ws](i❤️.ws), which does not feature a redirect.

## Actual result:
Lacking proper support, the domain falls back to [cooldomain.ws](cooldomain.ws).

## Expected result:
😎🌐.ws

- Is the issue reproducible on the latest version of Chrome? 
yes

**Comment by zachoooo** [comment]
Is there any way to enable emojis being used? What about allowing display for specific websites? Google Chrome has created a whole algorithm for checking what domains to display normally and which to convert to punycode.

**Comment by bsclifton** [comment]
cc: @diracdeltas for security implications of this - not sure if we'd want to allow this. As a user, it would be cool - but it could be abused too

**Comment by bridiver** [comment]
@zachoooo do you have any examples that display differently on Chrome than Brave? @KiaraGrouwstra this is actually a feature, not a bug. The reason only some domains are displayed without Punycode is to prevent domain spoofing https://chromium.googlesource.com/chromium/src/+/main/docs/idn.md

**Comment by ferry-creator** [comment]
@bridiver yes, i think it makes a lot of sense in terms of preventing domain spoofing.

But i'm thinking, emojis are only a small subset of characters, would there be any way to make some kind of whitelist for emojis? Especially with Handshake TLD domains possibly showing up more frequently in the future, it would once again be another crypto-minded feature in brave-browser ✌️
